db/resync_status.go

```diff
@@ -0,0 +1,105 @@
+package db
+
+import (
+	"sync"
+)
+
+type ResyncManager struct {
+	Status     ResyncStatus
+	LastError  error
+	Terminator bool       // Allows resync operation to be cancelled while in progress
+	Mutex      sync.Mutex // Used to lock the Status, LastError and Terminator
```

Comment:
Sync Gateway generally uses the naming convention 'Lock' for mutexes.  I'd also expect mutexes to be private to the ResyncManager, so 'lock' in this case.